src/librustc/mir/cache.rs

```diff
@@ -55,7 +55,8 @@ impl Cache {
         *self.predecessors.borrow_mut() = None;
     }
 
-    pub fn predecessors(&self, mir: &Mir) -> ReadGuard<IndexVec<BasicBlock, Vec<BasicBlock>>> {
+    pub fn predecessors(&self, mir: &Mir<'_>)
+                        -> ReadGuard<'_, IndexVec<BasicBlock, Vec<BasicBlock>>> {
```

Comment:
This exact function confuses me every time — I always think "how come there is no lifetime here?!"